File: architecture/typescript/01-typescript-architecture-layered/src/persons/person.service.ts
import { Person } from './person.model'
import { PersonRepository } from './person.repository'

export class PersonService {
  constructor(private repository: PersonRepository) {}

  listPersons(): Person[] {
    return this.repository.findAll()
  }

  addPerson(input: Omit<Person, 'id'>): Person {
    return this.repository.add(input)
  }
}

File: springboot-java-21/09-springboot-crud-layered/src/main/java/com/ganatan/starter/modules/person/PersonService.java
package com.ganatan.starter.modules.person;

import org.springframework.stereotype.Service;

import java.util.Collection;

@Service
public class PersonService {

	private final PersonRepository repository;

	public PersonService(PersonRepository repository) {
		this.repository = repository;
	}

	public Collection<Person> getAll() {
		return repository.findAll();
	}

	public Person getById(Long id) {
		return repository.findById(id);
	}

	public Person create(Person person) {
		return repository.save(person);
	}

	public Person update(Long id, Person updated) {
		if (repository.existsById(id)) {
			updated.setId(id);
			return repository.save(updated);
		}
		return null;
	}

	public boolean delete(Long id) {
		if (repository.existsById(id)) {
			repository.delete(id);
			return true;
		}
		return false;
	}
}
